### PR TITLE
Finished ticket #5

### PR DESCRIPTION
**Pull request by blameaimer**
-Creating test case
-Refactoring and testing function

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -90,16 +91,18 @@ describe("/api/topics", () => {
                   body: expect.any(String),
                   topic: expect.any(String),
                   created_at: expect.any(String),
-                  votes: expect.any(Number)
+                  votes: expect.any(Number),
+                  count: expect.any(String)
                 })
+                expect(response.body.article.count).toBe("11")
           });
       });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:99]: You shouldn't amended this test!! You should be adding a new test to add this new functionality.
eg
```js
expect(articles[0].comment_count).toBe(11)
```

Comment (daviidbartlett) [reply __tests__/app.test.js:99]: if you have to change an old test, it's a sign it was bad test. It was brittle

File: models/article-model.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      const topic = rows[0];
+      if (!topic) {
+        return Promise.reject({
+          status: 404,
+          msg: `No topic found for id: ${id}`,
+        });
+      }
+      return topic;
+    })
+    .catch((err) => {
```
Comment (daviidbartlett) [inline models/article-model.js:24]: no catch blocks in your models!!

File: models/article-model.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+FROM articles 
+LEFT JOIN users ON articles.author = users.username
+ORDER BY title DESC
+`)
+    .then(({ rows }) => {
+      return rows;
+    })
+    .catch((err) => {
+      return Promise.reject(err);
+    });
```
Comment (daviidbartlett) [inline models/article-model.js:42]: no thanks

File: models/article-model.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        return Promise.reject({
+          status: 404,
+          msg: `No topic found for id: ${id}`,
+        });
+      }
+      return topic;
+    })
+    .catch((err) => {
+      return Promise.reject(err);
+    });
```
Comment (daviidbartlett) [inline models/article-model.js:67]: nope

**Review by maddierosecodes — APPROVED** [review]
:)

**Pull request merged**